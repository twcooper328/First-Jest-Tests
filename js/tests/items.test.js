const items = require('./items');

describe('testing a list of objects inside of an array', () => {
  it('has not been changed', () => {
    expect(items.items).toMatchSnapshot();

    expect(items.items).toEqual(
      expect.arrayContaining([
        {
          id: 2,
          name: "coat",
          size: "small",
          desc: "Looks like this coat is back in fashion"
        }
      ])
    )
  });

  it('has id properties', () => {
    expect(items.items).toEqual(
      expect.arrayContaining([        // going into the array
        expect.objectContaining({        // going into the object
          "id": 1
        })
      ])
    );

    expect(items.items).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          "id": 4
        })
      ])
    );
  });

  it('array of objects has the correct length', () => {
    expect(items.items).toHaveLength(4);
  });

  it('has the correct property names in order', () => {
    const itemNames = items.items.map(item => item.name);
    expect(itemNames).toEqual([
      'hat',
      'coat',
      'boots',
      'watch'
    ]);

    const itemsOutOfOrder = [
      'coat',
      'hat',
      'boots',
      'watch'
    ];

    expect(itemNames).not.toEqual(itemsOutOfOrder);

  })

  it('does not contain incorrect object', () => {
    const notExpected = {
      id: 5,
      name: 'pinky-ring',
      size: 'small',
      desc: 'makes a statement'
    }

    expect(items.items).toEqual(
      expect.not.arrayContaining([notExpected])
    )

  })

})
