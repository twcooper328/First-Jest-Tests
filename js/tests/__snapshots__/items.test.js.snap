// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing a list of objects inside of an array has not been changed 1`] = `
Array [
  Object {
    "desc": "A hat that reminds you of childhood",
    "id": 1,
    "name": "hat",
    "size": "large",
  },
  Object {
    "desc": "Looks like this coat is back in fashion",
    "id": 2,
    "name": "coat",
    "size": "small",
  },
  Object {
    "desc": "Tough but elegant",
    "id": 3,
    "name": "boots",
    "size": "medium",
  },
  Object {
    "desc": "Connects to my phone",
    "id": 4,
    "name": "watch",
    "size": "medium",
  },
]
`;
